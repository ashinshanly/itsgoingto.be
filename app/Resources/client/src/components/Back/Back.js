import React from 'react'
import { browserHistory } from 'react-router'
import './Back.scss'

export const Back = () => {
  const submit = () => {
    browserHistory.push('/')
  }

  return (
    <div className='container back-container'>
      <div className='back-button'>
        <a onClick={submit}>
          <i className='fa fa-arrow-left' /> New Poll
        </a>
      </div>
    </div>
  )
}

export default Back
