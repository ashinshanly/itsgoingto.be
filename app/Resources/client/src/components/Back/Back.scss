@import "bourbon";
@import "neat";
@import "variables";

.back-container {
  position: absolute;
  width: auto;
  top: 0px;
  left: 0px;
  z-index: 200;
  .back-button {
    a {
      line-height: 50px;
      margin-left: 10px;
      color: $dark-grey;
      cursor: pointer;
      font-size: 1rem;
      transition: color 0.4s ease;

      &:hover {
        color: $base-font-color;
      }
    }
  }
}
