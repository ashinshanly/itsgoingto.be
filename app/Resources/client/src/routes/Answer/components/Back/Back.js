import React from 'react'
import { browserHistory } from 'react-router'
import './Back.scss'

class Back extends React.Component {
  submit = () => {
    browserHistory.push('/react')
  }

  render = () => (
    <div className="container back-container">
      <div className="back-button">
        <a onClick={ this.submit }>
          <i className="fa fa-arrow-left"></i> New Poll
        </a>
      </div>
    </div>
  )
}

export default Back
