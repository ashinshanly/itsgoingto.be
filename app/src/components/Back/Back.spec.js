/* global expect, jest */
import React from 'react'
import { browserHistory } from 'react-router'
import { shallow } from 'enzyme'
import Back from './Back'

describe('(Component) Back', () => {
  const wrapper = shallow(<Back />)

  describe('(Render) Back', () => {
    it('renders as a .container', () => {
      expect(wrapper.name()).toBe('div')
      expect(wrapper.hasClass('container')).toBe(true)
    })

    it('Should render the icon and text', () => {
      wrapper.find('a').should.contain(<i className='fa fa-arrow-left' />)
      expect(wrapper.find('a').text()).toBe(' New Poll')
    })

    describe('clickEvent', () => {
      browserHistory.push = jest.fn()

      it('Should call submit', () => {
        wrapper.find('a').simulate('click')
        expect(browserHistory.push).toBeCalledWith('/')
      })
    })
  })
})
