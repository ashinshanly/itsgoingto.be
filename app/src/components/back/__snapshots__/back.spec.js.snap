// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) Back (Render) matches snapshot 1`] = `
<div
  className="container back-container"
>
  <div
    className="back-button"
  >
    <a
      onClick={[Function]}
    >
      <i
        className="fa fa-arrow-left"
      />
       New Poll
    </a>
  </div>
</div>
`;
